Modules/Quotas/test/Quotas.Infrastructure.Tests/Tests/Repositories/RelationshipsRepositoryTests.cs

```diff
@@ -0,0 +1,118 @@
+﻿using Backbone.Modules.Quotas.Domain.Aggregates.Identities;
+using Backbone.Modules.Quotas.Infrastructure.Persistence.Database;
+using Backbone.Modules.Quotas.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Relationships.Domain.Aggregates.Relationships;
+using Backbone.Modules.Relationships.Infrastructure.Persistence.Database;
+using Backbone.Tooling;
+using Backbone.UnitTestTools.TestDoubles.Fakes;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Xunit;
+
+using static Backbone.Modules.Relationships.Domain.Tests.TestHelpers.TestData;
+
+namespace Backbone.Modules.Quotas.Infrastructure.Tests.Tests.Repositories;
+public class RelationshipsRepositoryTests
+{
+    private readonly RelationshipsDbContext _relationshipsArrangeContext;
+    private readonly QuotasDbContext _actContext;
+
+    public RelationshipsRepositoryTests()
+    {
+        AssertionScope.Current.FormattingOptions.MaxLines = 1000;
+
+        var connection = FakeDbContextFactory.CreateDbConnection();
+        (_relationshipsArrangeContext, _, _) = FakeDbContextFactory.CreateDbContexts<RelationshipsDbContext>(connection);
+        (_, _, _actContext) = FakeDbContextFactory.CreateDbContexts<QuotasDbContext>(connection);
+
+        SystemTime.Set(DateTime.UtcNow);
+    }
+
+    [Fact]
+    public async Task Count_pending_relationships()
+    {
+        // Arrange
+        var relationships = new List<Relationship>()
+        {
+            CreatePendingRelationship(),
+            CreatePendingRelationship()
+        };
+        await _relationshipsArrangeContext.Relationships.AddRangeAsync(relationships);
+        await _relationshipsArrangeContext.SaveChangesAsync();
+
+        var repository = new RelationshipsRepository(_actContext);
+        const QuotaPeriod quotaPeriod = QuotaPeriod.Hour;
+
+        // Act
+        var count = await repository.Count(IDENTITY_1, quotaPeriod.CalculateBegin(), quotaPeriod.CalculateEnd(), CancellationToken.None);
+
+        // Assert
+        count.Should().Be(2);
+    }
+
+    [Fact]
+    public async Task Count_active_relationships()
+    {
+        // Arrange
+        var relationships = new List<Relationship>()
+        {
+            CreateActiveRelationship(),
+            CreateActiveRelationship(),
+            CreateActiveRelationship()
```

Comment:
As a reader, I wonder why in the test for pending relationships you need 2 relationships, while in the test for active relationships, you need 3 of them.
=> If there is no difference, in the implementation, there should be no difference in the test.

Reply:
I was thinking in a way that creation of pending and active relationships are not depended as two separate isolated scenarios. So it was just a random number that I choose, two pending and three active. My goal was to test are they correctly counted. But I get the now the point of view of a reader. Changes made.